koku/api/dump/views.py

```diff
@@ -0,0 +1,29 @@
+"""Views for DumpRequest."""
+
+from rest_framework import mixins, viewsets
+from rest_framework.permissions import AllowAny
+
+from api.dump.models import DumpRequest
+from api.dump.serializers import DumpRequestSerializer
+
+
+class DumpRequestViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Create, retrieve, or list dump requests."""
+
+    queryset = DumpRequest.objects.all()
+    serializer_class = DumpRequestSerializer
+    permission_classes = (AllowAny,)
```

Comment:
This is probably not something we'll want `AllowAny` on since it is customer data. Since we have RBAC around who can view things in the API, we should have RBAC for this.

Reply:
This is an area of the system that I've only started trying to figure out. What specific `permissions_classes` do we currently have that you think would make sense here? I saw `OpenShiftAccessPermission` and `AWSAccessPermission`, but they don't look like the right fit. Should I make a new permission class?

(I may be totally wrong, though, because I don't yet understand those classes and their use cases fully.)

FWIW, `DumpRequestSerializer.create` sets the currently authenticated user on the model and `DumpRequestViewSet.get_serializer` limits the results to the currently authenticated user making the request. Keep in mind also that there is no actual customer cost management data accessible through _this_ model+serializer+view. The `DumpRequest` (rename pending) only acts like a request ticket for an export to happen, and the user can check back later to see the state of the export process. I would expect the async export process itself (not part of this PR) to do appropriate RBAC filtering for its various data collection and filtering operations.